barretenberg/cpp/src/barretenberg/crypto/pedersen_commitment/pedersen_refactor.cpp

```diff
@@ -0,0 +1,63 @@
+// TODO(@zac-wiliamson #2341 rename to pedersen.cpp once we migrate to new hash standard)
+
+#include "./pedersen_refactor.hpp"
+#include "./convert_buffer_to_field.hpp"
+#include "barretenberg/common/serialize.hpp"
+#include "barretenberg/common/throw_or_abort.hpp"
+#include <iostream>
+#ifndef NO_OMP_MULTITHREADING
+#include <omp.h>
+#endif
+
+namespace crypto {
+
+/**
+ * @brief Given a vector of fields, generate a pedersen commitment using the indexed generators.
+ *
+ * @details This method uses `Curve::BaseField` members as inputs. This aligns with what we expect when creating
+ * grumpkin commitments to field elements inside a BN254 SNARK circuit.
+ * @param inputs
+ * @param hash_index
+ * @param generator_context
+ * @return Curve::AffineElement
+ */
+template <typename Curve>
+typename Curve::AffineElement pedersen_commitment_refactor<Curve>::commit_native(
+    const std::vector<Fq>& inputs, const size_t hash_index, const generator_data<Curve>* const generator_context)
+{
+    const auto generators = generator_context->conditional_extend(inputs.size() + hash_index);
+    Element result = Group::point_at_infinity;
+
+    for (size_t i = 0; i < inputs.size(); ++i) {
+        result += Element(generators.get(i, hash_index)) * static_cast<uint256_t>(inputs[i]);
```

Comment:
why does this require a static cast to uint256_t ?

Reply:
`Curve::Fq` represents the field that `Curve` is defined over (i.e. x/y coordinate field) and `Curve::Fr` is the field whose modulus = the group order of `Curve`.

The `Curve` we're working over here is a generic SNARK-friendly curve. i.e. the SNARK circuit is defined over a field equivalent to `Curve::Fq`.

This adds complexity when we wish to commit to SNARK circuit field elements, as these are members of `Fq` and *not* `Fr`.  

We cast to `uint256_t` in order to convert an element of `Fq` into an `Fr` element, which is the required type when performing scalar multiplications.

For the case where `Curve` is `Grumpkin`, we know that `Grumpkin::Fq < Grumpkin::Fr`, so we won't end up with overflow issues by doing this.

I will add the above as a comment to the code, and also add some compile-time checks that `Fq < Fr`.

Does this all make sense to you?